### PR TITLE
Bind documentation

### PR DESCRIPTION
**Pull request by choumx**
Partial for #6199.

First pass on `amp-bind` documentation. Will expand as development progresses.

/to @bpaduch /cc @dvoytenko @cramforce

### CODE REVIEW

File: extensions/amp-bind/amp-bind.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
+# <a name="amp-bind-"></a> `amp-bind`
```
Comment (bpaduch) [inline extensions/amp-bind/amp-bind.md]: remove "-" from "amp-bind-"

File: extensions/amp-bind/amp-bind.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<table>
+  <tr>
+    <td class="col-fourty"><strong>Description</strong></td>
+    <td>`amp-bind` allows elements to mutate in response to user actions or data changes via data binding and simple JS-like expressions.</td>
```
Comment (bpaduch) [inline extensions/amp-bind/amp-bind.md]: replace `amp-bind` with <code>amp-bind</code>

File: extensions/amp-bind/amp-bind.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+Once installed, `amp-bind` scans the DOM for bindings -- an element may have attributes, CSS classes or textContent bound to a JS-like expression (see [Expressions](#expressions) below).
+
+The **scope** is implicit, mutable document state that binding expressions may reference. The scope may be initialized with a new AMP component `<amp-bind-state>`. Changes to the scope happen through user actions, e.g. clicking a `<button>` or switching slides on an `<amp-carousel>`.
+
+A **digest** is an evaluation of all binding expressions. Since scope is mutable and expressions can reference the scope, the evaluated result of expressions may change over time. Bound elements are updated as a result of a digest.
+
+A simple example:
+
+```
```
Comment (bpaduch) [inline extensions/amp-bind/amp-bind.md]: add language, so it's ```html

File: extensions/amp-bind/amp-bind.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  <script type=”application/json”>{ message: “Hello World” }</script>
+</amp-bind-state>
+
+<p [text]=”foo.message”>Placeholder text</p>
+```
+
+1. `amp-bind` scans the DOM and finds the `<p>` element’s `[text]` binding
+2. During the next digest, `amp-bind` reevaluates the expression `foo.message`
+3. On the next frame, `amp-bind` updates the `<p>` element's textContent from "Placeholder text" to "Hello World"
+
```
Comment (bpaduch) [inline extensions/amp-bind/amp-bind.md]: Add ending punctuation to these numbered steps.

File: extensions/amp-bind/amp-bind.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+`amp-bind` supports binding to three types of data on an element:
+
+| Type | Syntax | Details |
+| --- | --- | --- |
+| Attribute | `[<attr>]` | Only whitelisted attributes are supported.<br>Boolean expression results toggle boolean attributes.
+| CSS Classes | `[class]` | Expression result must be a space-delimited string.
+| Node.textContent | `[text]` | For applicable elements.
+
+Caveats:
```
Comment (bpaduch) [inline extensions/amp-bind/amp-bind.md]: Make Caveats bold.

File: extensions/amp-bind/amp-bind.md (hunk 1 above)
Comment (choumx) [inline extensions/amp-bind/amp-bind.md]: Done.

File: extensions/amp-bind/amp-bind.md (hunk 2 above)
Comment (choumx) [inline extensions/amp-bind/amp-bind.md]: ?

File: extensions/amp-bind/amp-bind.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+    <td class="col-fourty"><strong>Examples</strong></td>
+    <td>TBD</td>
+  </tr>
+</table>
+
+## Overview
+
+Once installed, `amp-bind` scans the DOM for bindings -- an element may have attributes, CSS classes or textContent bound to a JS-like expression (see [Expressions](#expressions) below).
+
+The **scope** is implicit, mutable document state that binding expressions may reference. The scope may be initialized with a new AMP component `<amp-bind-state>`. Changes to the scope happen through user actions, e.g. clicking a `<button>` or switching slides on an `<amp-carousel>`.
```
Comment (choumx) [inline extensions/amp-bind/amp-bind.md]: Done.

File: extensions/amp-bind/amp-bind.md (hunk 3 above)
Comment (choumx) [inline extensions/amp-bind/amp-bind.md]: Done.

File: extensions/amp-bind/amp-bind.md (hunk 4 above)
Comment (choumx) [inline extensions/amp-bind/amp-bind.md]: Done.

File: extensions/amp-bind/amp-bind.md (hunk 5 above)
Comment (choumx) [inline extensions/amp-bind/amp-bind.md]: Done.

File: extensions/amp-bind/amp-bind.md (hunk 2 above)
Comment (bpaduch) [inline extensions/amp-bind/amp-bind.md]: Github editor rendered what I wrote.  Please replace the backticks with `<code></code>`

Comment (choumx) [inline extensions/amp-bind/amp-bind.md]: Ah, got it. Done!

**Pull request merged**